dataprofiler/data_readers/data.py

```diff
@@ -53,6 +54,9 @@ def __new__(cls, input_file_path=None, data=None, data_type=None, options=None):
         if not options:
             options = dict()
 
+        if isinstance(input_file_path, str) and is_valid_url(input_file_path):
```

Comment:
i think the string check should be in the `is_valid_url` as oppose to here as the `is_valid_url` will fail if anything but a string is passed to it.